Sources/StreamChatUI/ChatMessageList/RETHINK/MessageContentView.swift

```diff
@@ -0,0 +1,501 @@
+        }
+
+        // Metadata
+        metadataView?.content = content.map {
+            .init(message: $0, isAuthorNameShown: layoutOptions?.contains(.authorName) == true)
+        }
+
+        // Quoted message view
+        quotedMessageView?.content = content?.quotedMessage.map {
+            .init(message: $0, avatarAlignment: $0.isSentByCurrentUser ? .right : .left)
+        }
+
+        // Thread info
+        if let replyCount = content?.replyCount {
+            threadReplyCountButton?.setTitle(L10n.Message.Threads.count(replyCount), for: .normal)
+        } else {
+            threadReplyCountButton?.setTitle(L10n.Message.Threads.reply, for: .normal)
+        }
+        let latestReplyAuthorAvatar = content?.latestReplies.first?.author.imageURL
+        threadAvatarView?.imageView.loadImage(from: latestReplyAuthorAvatar)
+
+        // Reactions view
+        reactionsBubbleView?.tailDirection = content
+            .map { $0.isSentByCurrentUser ? .toTrailing : .toLeading }
+        reactionsView?.content = content.map {
+            .init(
+                useBigIcons: false,
+                reactions: $0.reactions,
+                didTapOnReaction: nil
+            )
+        }
+    }
+
+    /// Cleans up the view so it is ready to display another message.
+    func prepareForReuse() {
+        content = nil
+        delegate = nil
+        indexPath = nil
+    }
+
+    // MARK: - Actions
+
+    /// Handles tap on `errorIndicatorView` and forwards the action to the delegate.
+    @objc open func handleTapOnErrorIndicator() {
+        guard let indexPath = indexPath else { return }
+
+        delegate?.didTapOnErrorIndicator(at: indexPath)
+    }
+
+    /// Handles tap on `threadReplyCountButton` and forwards the action to the delegate.
+    @objc open func handleTapOnThread() {
+        guard let indexPath = indexPath else { return }
+
+        delegate?.didTapOnThread(at: indexPath)
+    }
+
+    /// Handles tap on `quotedMessageView` and forwards the action to the delegate.
+    @objc open func handleTapOnQuotedMessage() {
+        guard let indexPath = indexPath else { return }
+
+        delegate?.didTapOnQuotedMessage(at: indexPath)
+    }
+}
+
+// MARK: - Setups
+
+private extension _MessageContentView {
+    func createTextView() -> UITextView {
+        if textView == nil {
+            textView = OnlyLinkTappableTextView().withoutAutoresizingMaskConstraints
+            textView?.isEditable = false
+            textView?.dataDetectorTypes = .link
+            textView?.isScrollEnabled = false
+            textView?.backgroundColor = .clear
+            textView?.adjustsFontForContentSizeCategory = true
+            textView?.textContainerInset = .init(top: 0, left: 8, bottom: 0, right: 8)
+            textView?.textContainer.lineFragmentPadding = 0
+            textView?.translatesAutoresizingMaskIntoConstraints = false
+            textView?.font = uiConfig.font.body
+        }
+        return textView!
+    }
```

Comment:
It seems to me these methods should rather be `open` than `private`, WDYT? Or is there a strong reason for not doing that?

Reply:
Yeah, missed that part. Let's open it 👍

Reply:
✅ Done!